tests/post/steps/test_volume.py

```diff
@@ -303,6 +350,45 @@ def _check_deletion_timestamp():
         name='checking the deletion timestamp of Volume {}'.format(name)
     )
 
+
+@then(parsers.parse("a pod '{name}' exist with file '{file}'"))
+def fetch_file_on_pod_volume(host, name, file, k8s_client):
+    def _check_pod_status():
+        pod = k8s_client.read_namespaced_pod(
+            name=name,
+            namespace='monitoring'
+        )
+        if pod is None:
+            assert False, 'Pod {} does not exist'.format(name)
+        try:
+            assert pod.status.phase == 'Running',\
+                'Unexpected status: expected Running, got {}'.format(
+                    pod.status.phase
+                )
+        except KeyError as exc:
+            assert False, 'Unexpected key error {} for pod {}'.format(
+                exc,
+                name
+            )
+    utils.retry(
+        _check_pod_status, times=30, wait=2,
+        name='checking status of Pod {}'.format(name)
+    )
+    with host.sudo():
```

Comment:
I don't think you need to have a retry for the pod availability in a retry.

Just have a retry around the code that check the for the file.
Also, we may want to check file using the API instead of `kubectl` + we also want to check the content of the file.

Reply:
Not sure, i can think of a better way to check the file using the API and it's content.

Can you enlighten me here??

Reply:
Take a look in `tests/install/steps/test_expansion.py`.

We're running an `etcdctl` command from inside the `etcd` container and get the output.

```python
etcd_member_list = etcdctl(k8s_client, ['member', 'list'], ssh_config)
[…]
def etcdctl(k8s_client, command, ssh_config):
    """Run an etcdctl command inside the etcd container."""
    name = 'etcd-{}'.format(
        utils.resolve_hostname('bootstrap', ssh_config)
    )
    etcd_command = [
        'etcdctl',
        '--endpoints', 'https://localhost:2379',
        '--ca-file', '/etc/kubernetes/pki/etcd/ca.crt',
        '--key-file', '/etc/kubernetes/pki/etcd/server.key',
        '--cert-file', '/etc/kubernetes/pki/etcd/server.crt',
    ] + command
    output = k8s.stream.stream(
        k8s_client.connect_get_namespaced_pod_exec,
        name=name, namespace='kube-system',
        command=etcd_command,
        stderr=True, stdin=False, stdout=True, tty=False
    )
    return output
```